fluss-lake/fluss-lake-iceberg/src/test/java/com/apache/fluss/lake/iceberg/catalog/IcebergLakeCatalogTest.java

```diff
@@ -0,0 +1,334 @@
+                flussIcebergCatalog.convertToIcebergSchema(tableDescriptor);
+
+        PartitionSpec expectedPartitionSpec =
+                flussIcebergCatalog.createPartitionSpec(tableDescriptor, icebergSchema);
+
+        SortOrder expectedSortOrder =
+                SortOrder.builderFor(icebergSchema).asc(OFFSET_COLUMN_NAME).build();
+
+        Map<String, String> tableProperties = new HashMap<>();
+        tableProperties.put("write.merge.mode", "merge-on-read");
+
+        Table createdTable =
+                icebergCatalog
+                        .buildTable(tableId, icebergSchema)
+                        .withPartitionSpec(expectedPartitionSpec)
+                        .withSortOrder(expectedSortOrder)
+                        .withProperties(tableProperties)
+                        .create();
+
+        assertThat(createdTable.spec().fields()).hasSize(1);
+        assertThat(createdTable.spec().fields().get(0).name()).isEqualTo("user_id_bucket");
+        assertThat(createdTable.spec().fields().get(0).transform().toString())
+                .contains("bucket[4]");
+
+        assertThat(createdTable.sortOrder().fields()).hasSize(1);
+        assertThat(createdTable.sortOrder().fields().get(0).sourceId())
+                .isEqualTo(icebergSchema.findField(OFFSET_COLUMN_NAME).fieldId());
+
+        assertThat(createdTable.properties()).containsEntry("write.merge.mode", "merge-on-read");
+    }
+
+    @Test
+    void rejectsPrimaryKeyTableWithMultipleBucketKeys() {
+        String database = "test_db";
+        String tableName = "multi_bucket_pk_table";
+
+        createNamespaceIfAbsent(database);
+
+        Schema flussSchema =
+                Schema.newBuilder()
+                        .column("user_id", DataTypes.BIGINT())
+                        .column("shop_id", DataTypes.BIGINT())
+                        .column("order_id", DataTypes.BIGINT())
+                        .column("amount", DataTypes.DOUBLE())
+                        .primaryKey("user_id", "shop_id") // Composite primary key
+                        .build();
+
+        TableDescriptor tableDescriptor =
+                TableDescriptor.builder()
+                        .schema(flussSchema)
+                        .distributedBy(4, "user_id", "shop_id") // Multiple bucket keys
+                        .build();
+
+        TableIdentifier tableId = TableIdentifier.of(database, tableName);
+
+        assertThatThrownBy(
+                        () -> {
+                            org.apache.iceberg.Schema icebergSchema =
+                                    flussIcebergCatalog.convertToIcebergSchema(tableDescriptor);
+                            PartitionSpec partitionSpec =
+                                    flussIcebergCatalog.createPartitionSpec(
+                                            tableDescriptor, icebergSchema);
+                            icebergCatalog.createTable(tableId, icebergSchema, partitionSpec);
```

Comment:
dito: we don't care about iceberg's catalog can create table or not since it's iceberg implementation.